packages/core/src/VRange.ts

```diff
@@ -22,6 +24,15 @@ export class VRange {
      * when necessary.
      */
     temporary: boolean;
+    /**
+     * Modifiers applied to the range will affect the rendering of range.start.
+     * It works as a sort of a cache for the current editing modifiers. Using
+     * the `modifiers` property of the start node would be too dangerous as any
+     * command might erroneously push modifiers in there without realizing they
+     * just messed with the cache. This value is reset each time the range
+     * changes in a document.
+     */
+    modifiers: Modifiers | undefined | null;
```

Comment:
`undefined | null` ?

Reply:
I copied the definition of the cache that this replaces. They really mean what they're called : undefined means the modifiers are undefined, and null that they're null. If it's undefined we need to recompute them, if it's null 'no modifier' replaces whatever modifier was there. The mechanism didn't change.